buildSrc/src/main/kotlin/io/spine/gradle/internal/Repositories.kt

```diff
@@ -0,0 +1,137 @@
+        val log = project.logger
+        log.info("Using credentials from `$credentialsFile`.")
+        val file = project.rootProject.file(credentialsFile)
+        if (!file.exists()) {
+            return null
+        }
+        val creds = file.readCredentials()
+        log.info("Publishing build as `${creds.username}`.")
+        return creds
+    }
+
+    private fun File.readCredentials(): Credentials {
+        val properties = Properties()
+        properties.load(inputStream())
+        val username = properties.getProperty("user.name")
+        val password = properties.getProperty("user.password")
+        return Credentials(username, password)
+    }
+
+    override fun toString(): String {
+        return name
+    }
+}
+
+/**
+ * Password credentials for a Maven repository.
+ */
+data class Credentials(
+    val username: String?,
+    val password: String?
+)
+
+/**
+ * Repositories to which we may publish. Normally, only one repository will be used.
+ *
+ * See `publish.gradle` for details of the publishing process.
+ */
+object PublishingRepos {
+
+    @Suppress("HttpUrlsUsage") // HTTPS is not supported by this repository.
+    val mavenTeamDev = Repository(
+        name = "maven.teamdev.com",
+        releases = "http://maven.teamdev.com/repository/spine",
+        snapshots = "http://maven.teamdev.com/repository/spine-snapshots",
+        credentialsFile = "credentials.properties"
+    )
+    val cloudRepo = Repository(
+        name = "CloudRepo",
+        releases = "https://spine.mycloudrepo.io/public/repositories/releases",
+        snapshots = "https://spine.mycloudrepo.io/public/repositories/snapshots",
+        credentialsFile = "cloudrepo.properties"
+    )
+
+    fun gitHub(repoName: String): Repository {
+        return Repository(
+            name = "GitHub Packages",
+            releases = "https://maven.pkg.github.com/SpineEventEngine/$repoName",
+            snapshots = "https://maven.pkg.github.com/SpineEventEngine/$repoName",
+            credentials = Credentials(
+                username = System.getenv("GITHUB_ACTOR"),
+                // This is a trick. Gradle only supports password or AWS credentials. Thus,
+                // we pass the GitHub token as a "password".
+                // https://docs.github.com/en/actions/guides/publishing-java-packages-with-gradle#publishing-packages-to-github-packages
+                password = System.getenv("GITHUB_TOKEN")
+            )
+        )
+    }
+}
+
+// Specific repositories.
+@Suppress("unused")
+object Repos {
+    val oldSpine: String = PublishingRepos.mavenTeamDev.releases
+    val oldSpineSnapshots: String = PublishingRepos.mavenTeamDev.snapshots
+
+    val spine: String = PublishingRepos.cloudRepo.releases
+    val spineSnapshots: String = PublishingRepos.cloudRepo.snapshots
+
+    const val sonatypeSnapshots: String = "https://oss.sonatype.org/content/repositories/snapshots"
+    const val gradlePlugins = "https://plugins.gradle.org/m2/"
```

Comment:
Can we use `gradlePluginPortal()` instead of our custom URL?

Reply:
It is already used in some of the places. This one is to be addressed sometime later. If you'd like to contribute to improving the `config`, please pitch in.